ui/src/org/pentaho/di/ui/cluster/dialog/SlaveServerDialog.java

```diff
@@ -525,35 +525,38 @@ public void ok() {
       if ( DialogUtils.objectWithTheSameNameExists( slaveServer, existingServers ) ) {
         String title = BaseMessages.getString( PKG, "SlaveServerDialog.SlaveServerNameExists.Title" );
         String message =
-            BaseMessages.getString( PKG, "SlaveServerDialog.SlaveServerNameExists", slaveServer.getName() );
+          BaseMessages.getString( PKG, "SlaveServerDialog.SlaveServerNameExists", slaveServer.getName() );
         String okButton = BaseMessages.getString( PKG, "System.Button.OK" );
         MessageDialog dialog =
-            new MessageDialog( shell, title, null, message, MessageDialog.ERROR, new String[] { okButton }, 0 );
+          new MessageDialog( shell, title, null, message, MessageDialog.ERROR, new String[] { okButton }, 0 );
 
         dialog.open();
         return;
       }
     }
 
-    originalServer.setName( slaveServer.getName() );
-    originalServer.setHostname( slaveServer.getHostname() );
-    originalServer.setPort( slaveServer.getPort() );
-    originalServer.setWebAppName( slaveServer.getWebAppName() );
-    originalServer.setUsername( slaveServer.getUsername() );
-    originalServer.setPassword( slaveServer.getPassword() );
+    synchronized ( originalServer ) {
```

Comment:
this syncronized looks useless.
it uses object's monitor to lock whereas the object's methods use readwrite lock